src/SprykerEco/Zed/Payone/Business/Payment/MethodSender/PayoneRefundMethodSender.php

```diff
@@ -0,0 +1,133 @@
+     * @param \SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReader $paymentMapperReader
+     * @param \Generated\Shared\Transfer\PayoneStandardParameterTransfer $standardParameter
+     * @param \SprykerEco\Zed\Payone\Business\Distributor\OrderPriceDistributorInterface $orderPriceDistributor
+     * @param \SprykerEco\Zed\Payone\Business\Payment\DataMapper\StandartParameterMapperInterface $standartParameterMapper
+     * @param \SprykerEco\Zed\Payone\Business\Payment\DataMapper\PayoneRequestProductDataMapperInterface $payoneRequestProductDataMapper
+     */
+    public function __construct(
+        AdapterInterface $executionAdapter,
+        PayoneQueryContainerInterface $queryContainer,
+        PaymentMapperReader $paymentMapperReader,
+        PayoneStandardParameterTransfer $standardParameter,
+        OrderPriceDistributorInterface $orderPriceDistributor,
+        StandartParameterMapperInterface $standartParameterMapper,
+        PayoneRequestProductDataMapperInterface $payoneRequestProductDataMapper
+    ) {
+        parent::__construct($queryContainer, $paymentMapperReader);
+        $this->executionAdapter = $executionAdapter;
+        $this->standardParameter = $standardParameter;
+        $this->orderPriceDistributor = $orderPriceDistributor;
+        $this->standartParameterMapper = $standartParameterMapper;
+        $this->payoneRequestProductDataMapper = $payoneRequestProductDataMapper;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PayoneRefundTransfer $refundTransfer
+     *
+     * @return \Generated\Shared\Transfer\RefundResponseTransfer
+     */
+    public function refundPayment(PayoneRefundTransfer $refundTransfer): RefundResponseTransfer
+    {
+        $distributedPriceOrderTransfer = $this->orderPriceDistributor->distributeOrderPrice(
+            $refundTransfer->getOrder()
+        );
+        $refundTransfer->setOrder($distributedPriceOrderTransfer);
+
+        $payonePaymentTransfer = $refundTransfer->getPayment();
+
+        $paymentEntity = $this->getPaymentEntity($payonePaymentTransfer->getFkSalesOrder());
+        $paymentMethodMapper = $this->getPaymentMethodMapper($paymentEntity);
+        $requestContainer = $paymentMethodMapper->mapPaymentToRefund($paymentEntity);
+        $requestContainer->setAmount(0 - $paymentEntity->getSpyPaymentPayoneDetail()->getAmount());
+        $requestContainer = $this->payoneRequestProductDataMapper->mapData($refundTransfer->getOrder(), $requestContainer);
+
+        $this->standartParameterMapper->setStandardParameter($requestContainer, $this->standardParameter);
+
+        $apiLogEntity = $this->initializeApiLog($paymentEntity, $requestContainer);
+
+        $rawResponse = $this->executionAdapter->sendRequest($requestContainer);
+        $responseContainer = new RefundResponseContainer($rawResponse);
+
+        $this->updateApiLogAfterRefund($apiLogEntity, $responseContainer);
+
+        $responseMapper = new RefundResponseMapper();
```

Comment:
* Extract to factory and use as normal dependency.

Reply:
Need discussion

Reply:
++